### PR TITLE
Add some more precises TypeScript interfaces

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Some records needs to be typed with some interfaces to be more strictly typed
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Can be tested by a dev if you build new-theme folder because it's mainly about build time errors
| Possible impacts? | BO migrated pages

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26022)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/pages/order/create/cart-editor.ts [hunk 1]
```diff
@@ -30,6 +30,20 @@ import createOrderMap from './create-order-map';
 
 const {$} = window;
 
+export interface CartAddress {
```
Comment (atomiix) [inline admin-dev/themes/new-theme/js/pages/order/create/cart-editor.ts]: Maybe to better differentiate this class from the one in `admin-dev/themes/new-theme/js/pages/order/create/create-order-page.ts`
```suggestion
export interface CartAddressIds {
```

**Comment by PierreRambaud** [comment]
Hello there, a little rebase? :D

**Review by Progi1984 — COMMENTED** [review]
A rebase, a small comment from  @atomiix and it's approved

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by matks** [comment]
Good for me. Thank you @NeOMakinG

**Pull request merged**